python/tests/test_drawing.py

```diff
@@ -1907,12 +1915,32 @@ def test_known_svg_ts(self):
         )
         # Prettify the SVG code for easy inspection
         svg = xml.dom.minidom.parseString(svg).toprettyxml()
-        svg_fn = os.path.join(os.path.dirname(__file__), "data", "svg", "ts.svg")
+        svg_fn = pathlib.Path(__file__).parent / "data" / "svg" / "ts.svg"
         self.verify_basic_svg(svg, width=200 * ts.num_trees)
         with open(svg_fn, "rb") as file:
             expected_svg = file.read()
         self.assertXmlEquivalentOutputs(svg, expected_svg)
 
+    def test_known_svg_internal_samples_ts(self, internal_samples_examples):
```

Comment:
Re the last comment, what if we did
```
@pytest.mark.parametrize("ts", internal_samples_examples)
def  def test_known_svg_internal_samples_ts(self, ts):
...
```

Parametrising is much better because it tells you *which* example has failed and lets you make things much more modular.

Reply:
Yes, you're absolutely right.